activesupport/lib/active_support/hash_with_indifferent_access.rb

```diff
@@ -239,7 +239,7 @@ def values_at(*indices)
     #   hash.fetch_values('a', 'c') { |key| 'z' } # => ["x", "z"]
     #   hash.fetch_values('a', 'c') # => KeyError: key not found: "c"
     def fetch_values(*indices, &block)
-      indices.collect { |key| fetch(key, &block) }
+      super(*indices.map { |key| convert_key(key) }, &block)
```

Comment:
It should be safe to call `map!` here, due to the splat on `indices` (i.e. it will be a newly-allocated array).

Reply:
Yes you can same one object allocation, but `map!` needs significant more time.

Reply:
What makes you say that?  On my machine `map!` is consistently 10% faster than `map`.